### PR TITLE
Refactored to remove config.ExtraOptions

### PR DESCRIPTION
**Pull request by spowelljr**
Refactored the `--extra-config` flag to a string flag and removed the `config.ExtraConfig` object to prevent modifying a global variable.

### CODE REVIEW

File: cmd/minikube/cmd/start_flags.go [hunk 1]
```diff
@@ -402,6 +403,18 @@ func getCNIConfig(cmd *cobra.Command) string {
 	return chosenCNI
 }
 
+func getExtraOptions() config.ExtraOptionSlice {
```
Comment (medyagh) [inline cmd/minikube/cmd/start_flags.go:406]: add comment, say u mean kubernetes extra options

File: cmd/minikube/cmd/start_flags.go [hunk 2]
```diff
@@ -402,6 +403,18 @@ func getCNIConfig(cmd *cobra.Command) string {
...
+	extraOptionVals := config.ExtraOptionSlice{}
+	val := viper.GetString(extraOptions)
+	if val == "" {
+		return extraOptionVals
+	}
+	if err := extraOptionVals.Set(val); err != nil {
```
Comment (klaases) [inline cmd/minikube/cmd/start_flags.go:412]: Will this be able to handle multiple flags/options?

Or, if an error occurs, will all the options be ignored?

Comment (klaases) [reply cmd/minikube/cmd/start_flags.go:412]: Update: please include the .Set() error message in the klog.Errorf().

File: cmd/minikube/cmd/start_flags.go (hunk 1 above)
Comment (spowelljr) [reply cmd/minikube/cmd/start_flags.go:407]: Done!

File: cmd/minikube/cmd/start_flags.go (hunk 2 above)
Comment (spowelljr) [reply cmd/minikube/cmd/start_flags.go:413]: Added the error message!

**Review by klaases — APPROVED** [review]

**Pull request merged**